src/lib/y2network/ip_address.rb

```diff
@@ -67,12 +62,18 @@ def from_string(str)
     def initialize(address, prefix = nil)
       @address = IPAddr.new(address)
       @prefix = prefix
-      @prefix ||= @address.ipv4? ? IPV4_DEFAULT_PREFIX : IPV6_DEFAULT_PREFIX
     end
 
     # Returns a string representation of the address
     def to_s
-      host? ? @address.to_s : "#{@address}/#{@prefix}"
+      prefix? ? "#{@address}/#{@prefix}" : @address.to_s
```

Comment:
maybe also document that output can omit prefix if prefix is not defined in constructor. Ideally in example in top level class documentation.